### PR TITLE
[SDL2] Initialize input driver lazily

### PR DESCRIPTION
**Pull request by xen2**
This is mostly to to avoid creating input and setting global SDL Windows input callback (SDL.AddEventWatch()) when not needed (we use our own SDL2 window and input code)

### CODE REVIEW

File: Source/OpenTK/Platform/SDL2/Sdl2Factory.cs [hunk 1]
```diff
@@ -34,7 +34,8 @@ namespace OpenTK.Platform.SDL2
 {
     class Sdl2Factory : PlatformFactoryBase
     {
-        readonly Sdl2InputDriver InputDriver = new Sdl2InputDriver();
+        private readonly object inputDriverLock = new object();
+        private Sdl2InputDriver inputDriver = new Sdl2InputDriver();
```
Comment (cra0zy) [inline Source/OpenTK/Platform/SDL2/Sdl2Factory.cs]: Shouldn't this be null?

Comment (xen2) [inline Source/OpenTK/Platform/SDL2/Sdl2Factory.cs]: Yes definitely, good catch!
Sorry, must have confused myself while doing the line-by-line commit picking...

**Comment by xen2** [comment]
Fixed the default value which was still being initialized.

**Comment by cra0zy** [comment]
Makes sense, marging!!!

**Pull request merged**